src/caja-desktop-window.c

```diff
@@ -289,17 +366,24 @@ realize (GtkWidget *widget)
     CajaDesktopWindowPrivate *details;
     window = CAJA_DESKTOP_WINDOW (widget);
     details = window->details;
+    GdkDisplay *display;
 
     /* Make sure we get keyboard events */
-    gtk_widget_set_events (widget, gtk_widget_get_events (widget)
-                           | GDK_KEY_PRESS_MASK | GDK_KEY_RELEASE_MASK);
-    /* Do the work of realizing. */
-    GTK_WIDGET_CLASS (caja_desktop_window_parent_class)->realize (widget);
+    display = gtk_widget_get_display (widget);
+    if (GDK_IS_X11_DISPLAY (display))
+        gtk_widget_set_events (widget, gtk_widget_get_events (widget)
+                               | GDK_KEY_PRESS_MASK | GDK_KEY_RELEASE_MASK);
 
-    /* This is the new way to set up the desktop window */
-    set_wmspec_desktop_hint (gtk_widget_get_window (widget));
+    /*Do the work of realizing. */
+    GTK_WIDGET_CLASS (caja_desktop_window_parent_class)->realize (widget);
 
-    set_desktop_window_id (window, gtk_widget_get_window (widget));
+    /* This is the new way to set up the desktop window in x11 but not for wayland */
+    display = gtk_widget_get_display (widget);
+    if (GDK_IS_X11_DISPLAY (display))
+        {
+            set_wmspec_desktop_hint (gtk_widget_get_window (widget));
+            set_desktop_window_id (window, gtk_widget_get_window (widget));
+        }
```

Comment:
One level of indentation too much
```suggestion
    if (GDK_IS_X11_DISPLAY (display))
    {
        set_wmspec_desktop_hint (gtk_widget_get_window (widget));
        set_desktop_window_id (window, gtk_widget_get_window (widget));
    }
```

Reply:
Found and fixed, thanks